Fish 6772 (#235)
payara/Payara-Documentation: docs/modules/ROOT/pages/Technical Documentation/Ecosystem/IDE Integration/Eclipse Plugin/Transform to Jakarta-EE10.adoc

= Transform Maven Projects or Files From Java EE 8 to Jakarta EE 10

> :ordinal: 1

Reviewer: ```suggestion
```

> Using the `Payara Tools plugin`, you can transform whole projects or just a file from Java EE 8 to Jakarta EE 10.

Reviewer: ```suggestion
Using the `Payara Tools plugin`, you can transform whole projects or single files from Java EE 8 to Jakarta EE 10.
```

[[refactor-project]]
== Refactor a Java EE 8 Project to Jakarta EE 10
To refactor a Java EE 8 based *Maven* project right click on root project folder and choose *Payara Refactor->Transform to Jakarta EE 10*:

image::eclipse-plugin/transform-to-jakarta/root-folder-Jakarta-EE-10.png[Refactor Project]

The plugin will open a dialog to choose the destination folder where the new project will be created:

image::eclipse-plugin/transform-to-jakarta/choose-new-project-dest-folder.png[Choose Project Destination Folder]

After choosing the `Destination Directory`, a confirmation message will appear indicating that the project was generated.

image::eclipse-plugin/transform-to-jakarta/confirmation-folder.png[Confirmation Folder]

> IMPORTANT: You can see the output of the generation process in the Eclipse console.

Reviewer: ```suggestion
TIP: You can see the output of the generation process in the Eclipse console.
```

The Maven plugin that executes the transformation process will print its output to the IDE's console:

image::eclipse-plugin/transform-to-jakarta/mvn-package-folder.png[mvn package Folder]

A maven `BUILD SUCCESS` message should be logged in the IDE's console.

image::eclipse-plugin/transform-to-jakarta/build-success.png[Build Success]

> 

Reviewer: ```suggestion

NOTE: When the whole project is transformed, the plugin adds `JakartaEE10` at the end of the project name. In our example, `demo-JakartaEE10` project was created from `demo` project.

```

[[refactor-file]]

> == Transform a single file

Reviewer: ```suggestion
== Transform a Single File
```

To refactor a `Java EE 8 file` right click on it and choose *Payara Refactor->Transform to Jakarta EE 10* option:

image::eclipse-plugin/transform-to-jakarta/file-to-Jakarta-EE-10.png[Refactor File]

The plugin will open a dialog to choose the destination package where the new file will be created:

image::eclipse-plugin/transform-to-jakarta/choose-new-file-dest-folder.png[Choose File Destination Folder]

After choosing the `Destination Package`, a confirmation message will appear indicating that the file was generated.

image::eclipse-plugin/transform-to-jakarta/confirmation-file.png[Confirmation File]

> IMPORTANT: You can see the output of the generation process in the Eclipse console.

Reviewer: ```suggestion
TIP: You can see the output of the generation process in the Eclipse console.
```

The output of the Maven plugin that executes the transformation code can be reviewed in the IDE's console:

image::eclipse-plugin/transform-to-jakarta/mvn-package-file.png[mvn package File]

A `BUILD SUCCESS` message from the Maven plugin will be logged in the IDE's console.

image::eclipse-plugin/transform-to-jakarta/finish-file.png[Finisehd Message]

The new file will be located in the chosen destination package.

image::eclipse-plugin/transform-to-jakarta/file-created.png[File Created]

> NOTE: When the whole project is transformed, the plugin adds `JakartaEE10` at the end of the project name. In our example, `demo-JakartaEE10` project was created from `demo` project.

Reviewer: ```suggestion
```

Reviewer: Moved this to the project refactor section

> NOTE: When a single file is transformed, a jakartaee10 package containing the new file will be created in the destination package.

Reviewer: ```suggestion
NOTE: When a single file is transformed, a `jakartaee10` package containing the new file will be created in the destination package.
```
